src/App.tsx

```diff
@@ -21,55 +20,152 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
+type State = {
   goods: string[],
-  sortType: SortType,
+  isVisible: boolean,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  isSorted: boolean,
+  sortBy: SortType,
+};
 
-  // Sort and reverse goods if needed
-  // ...
+class App extends React.Component<{}, State> {
+  state:Readonly<State> = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    isSorted: false,
+    sortBy: SortType.NONE,
+  };
 
-  return visibleGoods;
-}
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({
+        goods: goodsFromServer,
+        isVisible: true,
+        isReversed: false,
+        isSorted: false,
+        sortBy: SortType.NONE,
+      });
+    }, 100000000000);
+  }
 
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+  visibleGoodsList = () => {
+    this.setState((state) => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  SortByLength = () => {
+    this.setState({ isSorted: true, sortBy: SortType.LENGTH });
+  };
+
+  SortByAlphabet = () => {
+    this.setState({ isSorted: true, sortBy: SortType.ALPABET });
+  };
+
+  Reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  Reset = () => {
+    this.setState(() => ({
+      sortBy: SortType.NONE,
+      isReversed: false,
+      isSorted: false,
+    }));
+  };
+
+  render():React.ReactNode {
+    const {
+      goods,
+      isVisible,
+      isReversed,
+      isSorted,
+      sortBy,
+    } = this.state;
+
+    const sortedGoods = [...goods];
+
+    if (isSorted) {
+      switch (sortBy) {
+        case SortType.LENGTH:
+          sortedGoods.sort((good, prevGood) => good.length - prevGood.length);
+          break;
+
+        case SortType.ALPABET:
+          sortedGoods.sort((good, prevGood) => good.localeCompare(prevGood));
+          break;
+
+        default:
+      }
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
```

Comment:
move this logic out of `render` method, render should be as as clear as possible